[2024-12-28T14:58:02.696+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-28T14:58:02.711+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: l30d_dump_pipeline.l30_dump_pipeline manual__2024-12-28T14:58:00.691280+00:00 [queued]>
[2024-12-28T14:58:02.719+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: l30d_dump_pipeline.l30_dump_pipeline manual__2024-12-28T14:58:00.691280+00:00 [queued]>
[2024-12-28T14:58:02.720+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 2
[2024-12-28T14:58:02.732+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): l30_dump_pipeline> on 2024-12-28 14:58:00.691280+00:00
[2024-12-28T14:58:02.739+0000] {standard_task_runner.py:72} INFO - Started process 3896 to run task
[2024-12-28T14:58:02.743+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'l30d_dump_pipeline', 'l30_dump_pipeline', 'manual__2024-12-28T14:58:00.691280+00:00', '--job-id', '245', '--raw', '--subdir', 'DAGS_FOLDER/L30D_data_dump.py', '--cfg-path', '/tmp/tmph4_nkt8m']
[2024-12-28T14:58:02.745+0000] {standard_task_runner.py:105} INFO - Job 245: Subtask l30_dump_pipeline
[2024-12-28T14:58:02.785+0000] {task_command.py:467} INFO - Running <TaskInstance: l30d_dump_pipeline.l30_dump_pipeline manual__2024-12-28T14:58:00.691280+00:00 [running]> on host 24435936fa97
[2024-12-28T14:58:02.858+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='haziq' AIRFLOW_CTX_DAG_ID='l30d_dump_pipeline' AIRFLOW_CTX_TASK_ID='l30_dump_pipeline' AIRFLOW_CTX_EXECUTION_DATE='2024-12-28T14:58:00.691280+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-28T14:58:00.691280+00:00'
[2024-12-28T14:58:02.861+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-12-28T14:58:02.862+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-12-28T14:58:02.863+0000] {logging_mixin.py:190} INFO - Current task name:l30_dump_pipeline state:running start_date:2024-12-28 14:58:02.712622+00:00
[2024-12-28T14:58:02.864+0000] {logging_mixin.py:190} INFO - Dag name:l30d_dump_pipeline and current dag run status:running
[2024-12-28T14:58:02.865+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-28T14:58:06.377+0000] {logging_mixin.py:190} INFO - <pyspark.sql.session.SparkSession object at 0x7f631bf36e30>
[2024-12-28T14:58:06.379+0000] {logging_mixin.py:190} INFO - Spark session created successfully
[2024-12-28T14:58:06.913+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/pipeline/pipeline.py", line 15, in l30d_data_dump_pipeline
    spark_l30d(topic, BROKER)
  File "/opt/airflow/etl/data_dump_v2.py", line 190, in spark_l30d
    .load()
  File "/opt/spark/python/pyspark/sql/readwriter.py", line 314, in load
    return self._df(self._jreader.load())
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
  File "/opt/spark/python/pyspark/errors/exceptions/captured.py", line 185, in deco
    raise converted from None
pyspark.errors.exceptions.captured.IllegalArgumentException: starting offset can't be latest for batch queries on Kafka
[2024-12-28T14:58:06.935+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-12-28T14:58:06.936+0000] {logging_mixin.py:190} INFO - Task start:2024-12-28 14:58:02.712622+00:00 end:2024-12-28 14:58:06.934731+00:00 duration:4.222109
[2024-12-28T14:58:06.938+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): l30_dump_pipeline> dag:<DAG: l30d_dump_pipeline> dagrun:<DagRun l30d_dump_pipeline @ 2024-12-28 14:58:00.691280+00:00: manual__2024-12-28T14:58:00.691280+00:00, state:running, queued_at: 2024-12-28 14:58:00.704622+00:00. externally triggered: True>
[2024-12-28T14:58:06.941+0000] {logging_mixin.py:190} INFO - Failure caused by starting offset can't be latest for batch queries on Kafka
[2024-12-28T14:58:06.943+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=l30d_dump_pipeline, task_id=l30_dump_pipeline, run_id=manual__2024-12-28T14:58:00.691280+00:00, execution_date=20241228T145800, start_date=20241228T145802, end_date=20241228T145806
[2024-12-28T14:58:06.963+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-28T14:58:06.965+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 245 for task l30_dump_pipeline (starting offset can't be latest for batch queries on Kafka; 3896)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/pipeline/pipeline.py", line 15, in l30d_data_dump_pipeline
    spark_l30d(topic, BROKER)
  File "/opt/airflow/etl/data_dump_v2.py", line 190, in spark_l30d
    .load()
  File "/opt/spark/python/pyspark/sql/readwriter.py", line 314, in load
    return self._df(self._jreader.load())
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
  File "/opt/spark/python/pyspark/errors/exceptions/captured.py", line 185, in deco
    raise converted from None
pyspark.errors.exceptions.captured.IllegalArgumentException: starting offset can't be latest for batch queries on Kafka
[2024-12-28T14:58:07.011+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-28T14:58:07.032+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-28T14:58:07.035+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
